src/routes/clients.js

```diff
@@ -61,14 +71,26 @@ router.post('/', (req, res) => {
       case 'name':
         errno = ERRNO_INVALID_API_CLIENT_NAME;
         break;
+      case 'permissions':
+        errno = ERRNO_INVALID_API_CLIENT_PERMISSION;
+        break;
       default:
         errno = ERRNO_BAD_REQUEST;
     }
     return ApiError(res, 400, errno, BAD_REQUEST);
   }
 
   db().then(models => {
-    models.Clients.create(req.body).then(client => {
+    let client;
+    models.sequelize.transaction(transaction => {
```

Comment:
The transaction is not committed/rolled back until the promise chain returned within the `models.sequelize.transaction` callback resolves/rejects.